dashboard/src/components/views/EnvironmentView/EventDetailsContent.js

```diff
@@ -39,11 +41,18 @@ class EventDetailsContent extends React.PureComponent {
         fetchPolicy="cache-and-network"
         variables={{ ...match.params, ns }}
       >
-        {({ client, data: { event } = {}, loading }) => {
-          if (!loading && (!event || event.deleted)) return <NotFoundView />;
+        {({ client, data: { event } = {}, loading, aborted }) => {
+          if (!loading && !aborted && (!event || event.deleted)) {
```

Comment:
`!loading && !aborted` and `loading || aborted ` seem ubiquitous across every use of the Query container. As a convenience what if we called the render prop with a extra prop. Eg.

```
// Query#render
const pending = loading || aborted;
children({ pending, aborted: false, ...props });

// Ex usage ...
<Query ...>
  ({ data, pending }) => {
    if (!pending && !data) {
      return <NotFoundView />;
    }
    
    return <Container loading={pending} ... />
  }}
</Query>
```

Reply:
We will want to render proper aborted states with a refetch button. This would keep the loading and pending wholly separate making a calculated pending state not worth it.